<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>msImpute: Imputation of peptide intensity by low-rank approximation • msImpute</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="msImpute: Imputation of peptide intensity by low-rank approximation">
<meta property="og:description" content="msImpute">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">msImpute</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.17</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/msImpute-vignette.html">msImpute: Imputation of peptide intensity by low-rank approximation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/DavisLaboratory/msImpute/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="msImpute-vignette_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>msImpute: Imputation of peptide intensity by low-rank approximation</h1>
                        <h4 class="author">Soroor Hediyeh-zadeh</h4>
            <address class="author_afil">
      The Walter and Eliza Hall Institute of Medical Research<br><h4 class="author">Andrew I. Webb</h4>
            <address class="author_afil">
      The Walter and Eliza Hall Institute of Medical Research<br><h4 class="author">Melissa J. Davis</h4>
            <address class="author_afil">
      The Walter and Eliza Hall Institute of Medical Research<br><small class="dont-index">Source: <a href="https://github.com/DavisLaboratory/msImpute/blob/master/vignettes/msImpute-vignette.Rmd"><code>vignettes/msImpute-vignette.Rmd</code></a></small>
      <div class="hidden name"><code>msImpute-vignette.Rmd</code></div>

    </address>
</address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>MsImpute is a package for imputation of peptide intensity in proteomics experiments. It additionally contains tools for MAR/MNAR diagnosis and assessment of distortions to the probability distribution of the data post imputation. This document covers case studies featuring data obtained by various Mass Spectrometry (MS) acquisition modes to demonstrate applications of tools implemented in the package. The datasets selected here represent data with MAR and left-censored MNAR missingness patterns. We demonstrate msImpute is able to correctly identify these patterns and inform user’s decisions in downstream analyses.</p>
    </div>
    
<div id="quick-start" class="section level1">
<h1 class="hasAnchor">
<a href="#quick-start" class="anchor"></a>Quick Start</h1>
<p>The package consists of the following main functions:</p>
<ul>
<li><p><code>selectFeatures</code>: identifies peptides with high biological dropout rate. These peptides have a high abundance and high dropout. Their missingness pattern is used as a diagnostic for MAR/MNAR missingness type, which can then inform later decisions around imputation of the data.</p></li>
<li><p><code>scaleData</code>: <em>msImpute</em> first scales the data before training a low-rank model</p></li>
<li><p><code>msImpute</code>: Main function that imputes missing values by learning a low-rank approximation of the data</p></li>
<li><p><code>findVariableFeatures</code>: finds peptide with high biological variance. We use this in <code><a href="../reference/computeStructuralMetrics.html">computeStructuralMetrics()</a></code></p></li>
<li><p><code>computeStructuralMetrics</code>: returns a number of metrics that measure distortions into the data after imputation.</p></li>
<li><p><code>plotCV2</code>: Plots the square of coefficient of variation versus average log-expression i.e. mean-CV^2 plot</p></li>
</ul>
<p>These functions overall are designed to inform user’s decision on picking an imputation strategy. For a more detailed workflow, please see <a href="https://github.com/soroorh/proteomicscasestudies/blob/master/msImputeUsersGuide.pdf">here</a></p>
</div>
<div id="tims-case-study-blood-plasma" class="section level1">
<h1 class="hasAnchor">
<a href="#tims-case-study-blood-plasma" class="anchor"></a>TIMS Case Study: Blood plasma</h1>
<p>The aim is to assess the missing patterns in ion mobility data by Prianichnikov et al. (2020), available from PXD014777. The <code>evidence</code> table of MaxQuant output was processed as described below. Rows are Modified Peptide IDs. Charge state variations are treated as distinct peptide species. For peptides with multiple identification types, the intensity is considered to be the median of reported intensity values. Reverse complements and contaminant peptides are discarded. Peptides with more than 4 observed intensity values are retained.</p>
<p>The data was acquired in two batches (over two days). We are interested to know if missing values are evenly distributed across batches, or there is a batch-specific dropout trend. The runs are also labeled by S1, S2 and S4 (source unknown). The aim is to use this information to work out if missing values occur due to technical or biological effects.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">msImpute</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://bioinf.wehi.edu.au/limma">limma</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">imputeLCMD</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/jokergoo/ComplexHeatmap">ComplexHeatmap</a></span>)
</pre></div>
<div id="data-processing" class="section level2">
<h2 class="hasAnchor">
<a href="#data-processing" class="anchor"></a>Data processing</h2>
<p>The following procedures were applied to process the data, which we later load from the package data.</p>
<div id="filter-by-detection" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-by-detection" class="anchor"></a>Filter by detection</h3>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="kw">pxd014777</span>)
<span class="kw">y</span> <span class="op">&lt;-</span> <span class="kw">pxd014777</span>
</pre></div>
<p>Zero values that will be converted to Inf/-Inf after log- transformation. Check if there are valid values in the data before log transformation</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/is.finite.html">is.infinite</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.matrix.html">data.matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(<span class="kw">y</span>))))
</pre></div>
<pre><code>## 
##  FALSE   TRUE 
## 610515    681</code></pre>
<p>There are zero values that will be converted to Inf/-Inf after log- transformation. Add a small offset to avoid infinite values:</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="kw">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(<span class="kw">y</span><span class="op">+</span><span class="fl">0.25</span>)
</pre></div>
</div>
<div id="normalization" class="section level3">
<h3 class="hasAnchor">
<a href="#normalization" class="anchor"></a>Normalization</h3>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="co"># quantile normalisation</span>
<span class="kw">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/normalizebetweenarrays.html">normalizeBetweenArrays</a></span>(<span class="kw">y</span>, method = <span class="st">"quantile"</span>)
</pre></div>
</div>
</div>
<div id="determine-missing-values-pattern" class="section level2">
<h2 class="hasAnchor">
<a href="#determine-missing-values-pattern" class="anchor"></a>Determine missing values pattern</h2>
<p>Determine dominant patterns of missing values by investigating the dropout pattern of high dropout peptides. We find top 500 peptides with higher than expected dropout rat and high average log-intensity, then make a heatmap of their dropout pattern.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="kw">hdp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectFeatures.html">selectFeatures</a></span>(<span class="kw">y</span>, n_features = <span class="fl">500</span>)
</pre></div>
<p><img src="msImpute-vignette_files/figure-html/unnamed-chunk-5-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="co"># construct matrix M to capture missing entries</span>
<span class="kw">M</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">y</span>),<span class="fl">1</span>,<span class="fl">0</span>)
<span class="kw">M</span> <span class="op">&lt;-</span> <span class="kw">M</span>[<span class="kw">hdp</span><span class="op">$</span><span class="kw">msImpute_feature</span>,]

<span class="co"># plot a heatmap of missingness patterns for the selected peptides</span>


<span class="kw">batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">"(2018.*)_RF.*"</span>,<span class="st">"\\1"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="kw">y</span>)))
<span class="kw">experiment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">".*(S[1-9]).*"</span>,<span class="st">"\\1"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="kw">y</span>)))
</pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="kw">ha_column</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/HeatmapAnnotation.html">HeatmapAnnotation</a></span>(batch = <span class="kw">batch</span>,
                               experiment = <span class="kw">experiment</span>,
                               col = <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(batch = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'20181023'</span> = <span class="st">"#B24745FF"</span>,
                                                    <span class="st">'20181024'</span>= <span class="st">"#00A1D5FF"</span>),
                                          experiment=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"S1"</span>=<span class="st">"#DF8F44FF"</span>,
                                                       <span class="st">"S2"</span>=<span class="st">"#374E55FF"</span>,
                                                       <span class="st">"S4"</span>=<span class="st">"#79AF97FF"</span>)))

<span class="kw">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html">Heatmap</a></span>(<span class="kw">M</span>,
column_title = <span class="st">"dropout pattern, columns ordered by dropout similarity"</span>,
              name = <span class="st">"Intensity"</span>,
              col = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"#8FBC8F"</span>, <span class="st">"#FFEFDB"</span>),
              show_row_names = <span class="fl">FALSE</span>,
              show_column_names = <span class="fl">FALSE</span>,
              cluster_rows = <span class="fl">TRUE</span>,
              cluster_columns = <span class="fl">TRUE</span>,
              show_column_dend = <span class="fl">FALSE</span>,
              show_row_dend = <span class="fl">FALSE</span>,
              top_annotation = <span class="kw">ha_column</span>,
              row_names_gp =  <span class="fu">gpar</span>(fontsize = <span class="fl">7</span>),
              column_names_gp = <span class="fu">gpar</span>(fontsize = <span class="fl">8</span>),
              heatmap_legend_param = <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="co">#direction = "horizontal",</span>
              heatmap_legend_side = <span class="st">"bottom"</span>,
              labels = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"observed"</span>,<span class="st">"missing"</span>),
              legend_width = <span class="fu">unit</span>(<span class="fl">6</span>, <span class="st">"cm"</span>)),
         )
<span class="kw">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html">draw</a></span>(<span class="kw">hm</span>, heatmap_legend_side = <span class="st">"left"</span>)
</pre></div>
<div class="figure" style="text-align: center">
<img src="msImpute-vignette_files/figure-html/unnamed-chunk-6-1.png" alt="Heatmap of missing value patterns for top 500 high dropout peptides" width="700"><p class="caption">
Heatmap of missing value patterns for top 500 high dropout peptides
</p>
</div>
<p>As it can be seen on the heatmap, missing values are evenly distributed over batches (there is no systematic clustering of samples by batch), however, samples labeled as <code>S2</code>in the file name tend to cluster together. It turns out that samples from <code>S1</code> and <code>S4</code> experiments are blood plasma from individuals, whereas samples labeled as <code>S2</code> are labeled as pooled plasma. This is a biological effect, and we just demonstrated that <code>msImpute</code> can detect biologically meaningful and informative results.</p>
</div>
</div>
<div id="dda-case-study-extracellular-vesicles-isolated-from-inflammatory-bowel-disease-patients-and-controls" class="section level1">
<h1 class="hasAnchor">
<a href="#dda-case-study-extracellular-vesicles-isolated-from-inflammatory-bowel-disease-patients-and-controls" class="anchor"></a>DDA Case Study: Extracellular vesicles isolated from inflammatory bowel disease patients and controls</h1>
<p>The study aims to characterize the proteomic profile of extracellular vesicles isolated from the descending colon of pediatric patients with inflammatory bowel disease and control participants. The following analysis is based on the <code>peptide</code> table from MaxQuant output, available from PXD007959. Rows are Modified Peptide IDs. Charge state variations are treated as distinct peptide species. Reverse complements and contaminant peptides are discarded. Peptides with more than 4 observed intensity values are retained. Additionally, qualified peptides are required to map uniquely to proteins. Two of the samples with missing group annotation were excluded.</p>
<div id="filter-by-detection-1" class="section level2">
<h2 class="hasAnchor">
<a href="#filter-by-detection-1" class="anchor"></a>Filter by detection</h2>
<p>The sample descriptions can be accessed via <code>pxd007959$samples</code>. Intensity values are stored in <code>pxd007959$y</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="kw">pxd007959</span>)

<span class="kw">sample_annot</span> <span class="op">&lt;-</span> <span class="kw">pxd007959</span><span class="op">$</span><span class="kw">samples</span>
<span class="kw">y</span> <span class="op">&lt;-</span> <span class="kw">pxd007959</span><span class="op">$</span><span class="kw">y</span>
<span class="kw">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(<span class="kw">y</span>)
</pre></div>
</div>
<div id="normalization-1" class="section level2">
<h2 class="hasAnchor">
<a href="#normalization-1" class="anchor"></a>Normalization</h2>
<p>We apply <code>cyclic loess</code> normalisation from <code>limma</code> to normalise log-intensities. We have justified use of <code>cyclic loess</code> method in depth in the user’s guide.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="kw">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/normalizebetweenarrays.html">normalizeBetweenArrays</a></span>(<span class="kw">y</span>, method = <span class="st">"cyclicloess"</span>)
</pre></div>
</div>
<div id="determine-missing-values-pattern-1" class="section level2">
<h2 class="hasAnchor">
<a href="#determine-missing-values-pattern-1" class="anchor"></a>Determine missing values pattern</h2>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="co"># determine missing values pattern</span>
<span class="kw">hdp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectFeatures.html">selectFeatures</a></span>(<span class="kw">y</span>, n_features = <span class="fl">500</span>)
</pre></div>
<p><img src="msImpute-vignette_files/figure-html/unnamed-chunk-9-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="co"># construct matrix M to capture missing entries</span>
<span class="kw">M</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">y</span>),<span class="fl">1</span>,<span class="fl">0</span>)
<span class="kw">M</span> <span class="op">&lt;-</span> <span class="kw">M</span>[<span class="kw">hdp</span><span class="op">$</span><span class="kw">msImpute_feature</span>,]



<span class="co"># plot a heatmap of missingness patterns for the selected peptides</span>
<span class="kw">ha_column</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/HeatmapAnnotation.html">HeatmapAnnotation</a></span>(group = <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="kw">sample_annot</span><span class="op">$</span><span class="kw">group</span>),
                               col=<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(group=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'Control'</span> = <span class="st">"#E64B35FF"</span>,
                                                <span class="st">'Mild'</span> = <span class="st">"#3C5488FF"</span>,
                                                <span class="st">'Moderate'</span> = <span class="st">"#00A087FF"</span>,
                                                <span class="st">'Severe'</span>=<span class="st">"#F39B7FFF"</span>)))

<span class="kw">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html">Heatmap</a></span>(<span class="kw">M</span>,
column_title = <span class="st">"dropout pattern, columns ordered by dropout similarity"</span>,
              name = <span class="st">"Intensity"</span>,
              col = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"#8FBC8F"</span>, <span class="st">"#FFEFDB"</span>),
              show_row_names = <span class="fl">FALSE</span>,
              show_column_names = <span class="fl">FALSE</span>,
              cluster_rows = <span class="fl">TRUE</span>,
              cluster_columns = <span class="fl">TRUE</span>,
              show_column_dend = <span class="fl">FALSE</span>,
              show_row_dend = <span class="fl">FALSE</span>,
              top_annotation = <span class="kw">ha_column</span>,
              row_names_gp =  <span class="fu">gpar</span>(fontsize = <span class="fl">7</span>),
              column_names_gp = <span class="fu">gpar</span>(fontsize = <span class="fl">8</span>),
              heatmap_legend_param = <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="co">#direction = "horizontal",</span>
              heatmap_legend_side = <span class="st">"bottom"</span>,
              labels = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"observed"</span>,<span class="st">"missing"</span>),
              legend_width = <span class="fu">unit</span>(<span class="fl">6</span>, <span class="st">"cm"</span>)),
         )
<span class="kw">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html">draw</a></span>(<span class="kw">hm</span>, heatmap_legend_side = <span class="st">"left"</span>)
</pre></div>
<div class="figure" style="text-align: center">
<img src="msImpute-vignette_files/figure-html/unnamed-chunk-10-1.png" alt="Dropout pattern of top 500 high dropout peptides" width="700"><p class="caption">
Dropout pattern of top 500 high dropout peptides
</p>
</div>
<p>As it can be seen, samples from the control group cluster together. There is a structured, block-wise pattern of missing values (i.e. missing values occur in all samples from control individuals, but are measured in disease group). This suggests that missing in not at random. This is an example of <strong>MNAR</strong> dataset. Given this knowledge, we impute using <code>QRILC</code> and <code>msImpute</code>. We then compare these methods by preservation of local (within experimental group) and global (between experimental group) similarities.</p>
</div>
<div id="imputation" class="section level2">
<h2 class="hasAnchor">
<a href="#imputation" class="anchor"></a>Imputation</h2>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="co"># imputation</span>

<span class="kw">y_qrilc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/imputeLCMD/man/impute.QRILC.html">impute.QRILC</a></span>(<span class="kw">y</span>)[[<span class="fl">1</span>]]

<span class="kw">y_msImpute</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scaleData.html">scaleData</a></span>(<span class="kw">y</span>)
</pre></div>
<pre><code>## bi-scaling ...
## data scaled</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="kw">y_msImpute</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/msImpute.html">msImpute</a></span>(<span class="kw">y_msImpute</span>)
</pre></div>
<pre><code>## maximum rank is 15 
## computing lambda0 ... 
## lambda0 is 44.79365 
## fit the low-rank model ... 
## model fitted. 
## Imputting missing entries ... 
## Imputation completed</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="kw">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="kw">sample_annot</span><span class="op">$</span><span class="kw">group</span>)
</pre></div>
</div>
<div id="assessment-of-preservation-of-local-and-global-structures" class="section level2">
<h2 class="hasAnchor">
<a href="#assessment-of-preservation-of-local-and-global-structures" class="anchor"></a>Assessment of preservation of local and global structures</h2>
<p>If you’ve installed python, and have set up a python environment in your session, you can run this section to compute the GW distance. Here we have just included the results returned by <code><a href="../reference/computeStructuralMetrics.html">computeStructuralMetrics()</a></code> as text, as this may not run on all systems, if the python environment is not setup beforehand. Please see the user’s guide for setup instructions. Note that you can still run <code><a href="../reference/computeStructuralMetrics.html">computeStructuralMetrics()</a></code>by setting <code>y=NULL</code>, if there are no python environments setup.</p>
<p><strong>Withinness, betweenness and Gromov-Wasserstein (GW) distance</strong></p>
<p><code>computeStructuralMerics</code> returns three metrics that can be used to compare various imputation procedures:</p>
<ul>
<li><p><code>withinness</code> is the sum of the squared distances between samples from the same experimental group (e.g. control, treatment, Het, WT). More specifically the similarity of the samples is measured by the distance of the (expression profile of the) sample from group centroid. This is a measure of preservation of local structures.</p></li>
<li><p><code>betweenness</code> is the sum of the squared distances between the experimental groups, more specifically the distance between group centroids. This is a measure of preservation of global structures.</p></li>
<li><p><code>gw_dist</code> is the Gromov-Wasserstein distance computed between Principal Components of imputed and source data. It is a measure of how well the structures are overall preserved over all principal axis of variation in the data. Hence, it captures preservation of both local and global structures. PCs of the source data are computed using highly variable peptides (i.e. peptides with high biological variance).</p></li>
</ul>
<p>An ideal imputation method results in smaller <code>withinness</code>, larger <code>withinness</code> and smaller <code>gw_dist</code> among other imputation methods.</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="kw">top.hvp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findVariableFeatures.html">findVariableFeatures</a></span>(<span class="kw">y</span>)

<span class="fu"><a href="../reference/computeStructuralMetrics.html">computeStructuralMetrics</a></span>(<span class="kw">y_msImpute</span>, <span class="kw">group</span>, <span class="kw">y</span>[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">top.hvp</span>)[<span class="fl">1</span><span class="op">:</span><span class="fl">50</span>],], k = <span class="fl">16</span>)
</pre></div>
<pre><code>Computing GW distance using k= 16 Principal Components
$withinness
    Mild  Control Moderate   Severe 
8.615751 9.455355 8.692705 8.586565 

$betweenness
[1] 9.358183

$gw_dist
[1] 0.04294529</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="fu"><a href="../reference/computeStructuralMetrics.html">computeStructuralMetrics</a></span>(<span class="kw">y_qrilc</span>, <span class="kw">group</span>, <span class="kw">y</span>[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">top.hvp</span>)[<span class="fl">1</span><span class="op">:</span><span class="fl">50</span>],], k = <span class="fl">16</span>)
</pre></div>
<pre><code>$withinness
    Mild  Control Moderate   Severe 
10.32911 11.83523 10.61441 10.76315 

$betweenness
[1] 11.61141

$gw_dist
[1] 0.008813667</code></pre>
<p><code>Withinness</code> is smaller by <code>msImpute</code>, which indicates that local structures are better preserved by these two methods. However, the <code>gw_dist</code> is smaller for <code>QRILC</code> over all PCs. This suggests that <code>QRILC</code> is likely a better approach compared to the other two methods. Given that the dominant patterns of missing values in this dataset is MNAR, <code>QRILC</code> is indeed a reasonable choice, as missing values are likely to be left-censored MNAR. Note that <code>k</code> is set to the number of samples to capture all dimensions of the data.</p>
</div>
</div>
<div id="swath-dia-case-study-swath-ms-analysis-of-gfi1-mutant-bone-marrow-neutrophils" class="section level1">
<h1 class="hasAnchor">
<a href="#swath-dia-case-study-swath-ms-analysis-of-gfi1-mutant-bone-marrow-neutrophils" class="anchor"></a>SWATH-DIA Case Study: SWATH-MS analysis of Gfi1-mutant bone marrow neutrophils</h1>
<p>This study investigates the proteomic alterations in bone marrow neutrophils isolated from 5-8 week old Gfi1+/-, Gfi1K403R/-, Gfi1R412X/-, and Gfi1R412X/R412X mice using the SWATH-MS technique. This dataset consists of 13 DIA (for SWATH) runs on a TripleTOF 5600 plus (SCIEX). Data available from PXD010943. Peak areas extracted from <code>13DIAs_SWATHprocessing_area_score_FDR_observedRT.xlsx</code>.<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a></p>
<p>Rows are peptides. Charge state variations are treated as distinct peptide species. Peptides with more than 4 observed intensity values are retained.</p>
<div id="normalization-2" class="section level3">
<h3 class="hasAnchor">
<a href="#normalization-2" class="anchor"></a>Normalization</h3>
<p>We normalize using <code>quantile normalization</code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="kw">pxd010943</span>)
<span class="kw">y</span> <span class="op">&lt;-</span> <span class="kw">pxd010943</span>
<span class="co"># no problematic values for log- transformation</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/is.finite.html">is.infinite</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.matrix.html">data.matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(<span class="kw">y</span>))))
</pre></div>
<pre><code>## 
## FALSE 
## 30641</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="kw">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(<span class="kw">y</span>)
<span class="kw">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/normalizebetweenarrays.html">normalizeBetweenArrays</a></span>(<span class="kw">y</span>, method = <span class="st">"quantile"</span>)
</pre></div>
</div>
<div id="determine-missing-values-pattern-2" class="section level2">
<h2 class="hasAnchor">
<a href="#determine-missing-values-pattern-2" class="anchor"></a>Determine missing values pattern</h2>
<p>We use top 100 high dropout peptides as there are 180 partially observed peptides overall</p>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="kw">hdp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectFeatures.html">selectFeatures</a></span>(<span class="kw">y</span>, n_features = <span class="fl">100</span>) 
</pre></div>
<p><img src="msImpute-vignette_files/figure-html/unnamed-chunk-15-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="co"># construct matrix M to capture missing entries</span>
<span class="kw">M</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">y</span>),<span class="fl">1</span>,<span class="fl">0</span>)
<span class="kw">M</span> <span class="op">&lt;-</span> <span class="kw">M</span>[<span class="kw">hdp</span><span class="op">$</span><span class="kw">msImpute_feature</span>,]

<span class="co"># plot a heatmap of missingness patterns for the selected peptides</span>

<span class="kw">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">"_[1234]"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="kw">y</span>)))

<span class="kw">group</span>
</pre></div>
<pre><code>##  [1] Gfi1      Gfi1      Gfi1      Gfi1      R412Xhet  R412Xhet  R412Xhet 
##  [8] R412Xhet  R412Xhomo R412Xhomo K403R     K403R     K403R    
## Levels: Gfi1 K403R R412Xhet R412Xhomo</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="kw">ha_column</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/HeatmapAnnotation.html">HeatmapAnnotation</a></span>(group = <span class="kw">group</span>)

<span class="kw">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html">Heatmap</a></span>(<span class="kw">M</span>,
column_title = <span class="st">"dropout pattern, columns ordered by dropout similarity"</span>,
              name = <span class="st">"Intensity"</span>,
              col = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"#8FBC8F"</span>, <span class="st">"#FFEFDB"</span>),
              show_row_names = <span class="fl">FALSE</span>,
              show_column_names = <span class="fl">FALSE</span>,
              cluster_rows = <span class="fl">TRUE</span>,
              cluster_columns = <span class="fl">TRUE</span>,
              show_column_dend = <span class="fl">FALSE</span>,
              show_row_dend = <span class="fl">FALSE</span>,
              top_annotation = <span class="kw">ha_column</span>,
              row_names_gp =  <span class="fu">gpar</span>(fontsize = <span class="fl">7</span>),
              column_names_gp = <span class="fu">gpar</span>(fontsize = <span class="fl">8</span>),
              heatmap_legend_param = <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="co">#direction = "horizontal",</span>
              heatmap_legend_side = <span class="st">"bottom"</span>,
              labels = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"observed"</span>,<span class="st">"missing"</span>),
              legend_width = <span class="fu">unit</span>(<span class="fl">6</span>, <span class="st">"cm"</span>)),
         )
<span class="kw">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html">draw</a></span>(<span class="kw">hm</span>, heatmap_legend_side = <span class="st">"left"</span>)
</pre></div>
<div class="figure" style="text-align: center">
<img src="msImpute-vignette_files/figure-html/unnamed-chunk-16-1.png" alt="Dropout pattern of top 100 high dropout peptides" width="700"><p class="caption">
Dropout pattern of top 100 high dropout peptides
</p>
</div>
<p>It can be seen that missing values are evenly distributed across the groups, and there is no systematic clustering of samples by experimental group. Hence, the dominant pattern of missingness is likely to be <strong>MAR</strong>.</p>
<p>We impute the data using a left-censored MNAR method (<code>QRILC</code>) and <code>msImpute</code> which is suitable for MAR datasets, then compare the two approaches.</p>
</div>
<div id="imputation-1" class="section level2">
<h2 class="hasAnchor">
<a href="#imputation-1" class="anchor"></a>Imputation</h2>
<div class="sourceCode" id="cb30"><pre class="downlit">
<span class="co"># imputation</span>
<span class="kw">y_qrilc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/imputeLCMD/man/impute.QRILC.html">impute.QRILC</a></span>(<span class="kw">y</span>)[[<span class="fl">1</span>]]
</pre></div>
<p>Note that the scaling algorithm in <code>msImpute</code> does not converge at default iterations in this case. The number of iterations is increased to achieve convergence.</p>
<div class="sourceCode" id="cb31"><pre class="downlit">
<span class="kw">y_msImpute</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scaleData.html">scaleData</a></span>(<span class="kw">y</span>, maxit = <span class="fl">50</span>) <span class="co"># an example where iteration does not converge at default iteration number</span>
</pre></div>
<pre><code>## bi-scaling ...
## data scaled</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit">
<span class="kw">y_msImpute</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/msImpute.html">msImpute</a></span>(<span class="kw">y_msImpute</span>)
</pre></div>
<pre><code>## maximum rank is 12 
## computing lambda0 ... 
## lambda0 is 113.9174 
## fit the low-rank model ... 
## model fitted. 
## Imputting missing entries ... 
## Imputation completed</code></pre>
</div>
<div id="assessment-of-preservation-of-local-and-global-structures-1" class="section level2">
<h2 class="hasAnchor">
<a href="#assessment-of-preservation-of-local-and-global-structures-1" class="anchor"></a>Assessment of preservation of local and global structures</h2>
<p>In this example, we do not compute <code>gw_dist</code> and only rely on <code>withinness</code> and <code>betweenness</code> metrics to assess imputation.</p>
<div class="sourceCode" id="cb35"><pre class="downlit">
<span class="kw">top.hvp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findVariableFeatures.html">findVariableFeatures</a></span>(<span class="kw">y</span>)
</pre></div>
<p><img src="msImpute-vignette_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<div class="sourceCode" id="cb36"><pre class="downlit">
<span class="fu"><a href="../reference/computeStructuralMetrics.html">computeStructuralMetrics</a></span>(<span class="kw">y_msImpute</span>, <span class="kw">group</span>, y=<span class="kw">NULL</span>)
</pre></div>
<pre><code>## $withinness
##      Gfi1  R412Xhet R412Xhomo     K403R 
##  8.892049  9.145670  7.279971  6.446897 
## 
## $betweenness
## [1] 9.87758</code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit">
<span class="fu"><a href="../reference/computeStructuralMetrics.html">computeStructuralMetrics</a></span>(<span class="kw">y_qrilc</span>, <span class="kw">group</span>, y = <span class="kw">NULL</span>)
</pre></div>
<pre><code>## $withinness
##      Gfi1  R412Xhet R412Xhomo     K403R 
##  9.040203  9.298123  7.487923  6.459766 
## 
## $betweenness
## [1] 10.01687</code></pre>
<p>Note <code>withinness</code> tend to be smaller in <code>msImpute</code>, while <code>betweenness</code> is marginally larger for <code>QRILC</code>. The differences in <code>betweenness</code> between <code>QRILC</code> and <code>msImpute</code> is small. Note that since such a small proportion of peptides are missing, and the dominant missingness pattern appear to be MAR, one can also choose to not impute depending on the downstream analyses.</p>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>Prianichnikov, N., Koch, H., Koch, S., Lubeck, M., Heilig, R., Brehmer, S., Fischer, R., &amp; Cox, J. (2020). MaxQuant Software for Ion Mobility Enhanced Shotgun Proteomics. Molecular &amp; cellular proteomics : MCP, 19(6), 1058–1069. <a href="https://doi.org/10.1074/mcp.TIR119.001720" class="uri">https://doi.org/10.1074/mcp.TIR119.001720</a></p>
<p>Zhang, X., Deeke, S.A., Ning, Z. et al. Metaproteomics reveals associations between microbiome and intestinal extracellular vesicle proteins in pediatric inflammatory bowel disease. Nat Commun 9, 2873 (2018). <a href="https://doi.org/10.1038/s41467-018-05357-4" class="uri">https://doi.org/10.1038/s41467-018-05357-4</a></p>
<p>Muench, D.E., Olsson, A., Ferchen, K. et al. Mouse models of neutropenia reveal progenitor-stage-specific defects. Nature 582, 109–114 (2020). <a href="https://doi.org/10.1038/s41586-020-2227-7" class="uri">https://doi.org/10.1038/s41586-020-2227-7</a></p>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<pre><code>## R version 4.0.2 Patched (2020-09-17 r79226)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.1 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-openmp/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] ComplexHeatmap_2.5.5 imputeLCMD_2.0       impute_1.63.0       
##  [4] pcaMethods_1.81.0    Biobase_2.49.1       BiocGenerics_0.35.4 
##  [7] norm_1.0-9.5         tmvtnorm_1.4-10      gmm_1.6-5           
## [10] sandwich_2.5-1       Matrix_1.2-18        mvtnorm_1.1-1       
## [13] limma_3.45.14        msImpute_0.99.17     BiocStyle_2.17.0    
## 
## loaded via a namespace (and not attached):
##  [1] edgeR_3.31.4                BiocSingular_1.5.0         
##  [3] DelayedMatrixStats_1.11.1   scuttle_0.99.13            
##  [5] assertthat_0.2.1            statmod_1.4.34             
##  [7] highr_0.8                   BiocManager_1.30.10        
##  [9] dqrng_0.2.1                 GenomeInfoDbData_1.2.3     
## [11] yaml_2.2.1                  backports_1.1.10           
## [13] lattice_0.20-41             digest_0.6.25              
## [15] GenomicRanges_1.41.6        RColorBrewer_1.1-2         
## [17] XVector_0.29.3              colorspace_1.4-1           
## [19] htmltools_0.5.0             pkgconfig_2.0.3            
## [21] GetoptLong_1.0.2            bookdown_0.20              
## [23] zlibbioc_1.35.0             cpp11_0.2.1                
## [25] BiocParallel_1.23.2         IRanges_2.23.10            
## [27] pdist_1.2                   SummarizedExperiment_1.19.6
## [29] magrittr_1.5                crayon_1.3.4               
## [31] memoise_1.1.0               evaluate_0.14              
## [33] fs_1.5.0                    bluster_0.99.1             
## [35] data.table_1.13.0           tools_4.0.2                
## [37] softImpute_1.4              GlobalOptions_0.1.2        
## [39] matrixStats_0.56.0          stringr_1.4.0              
## [41] S4Vectors_0.27.13           locfit_1.5-9.4             
## [43] cluster_2.1.0               DelayedArray_0.15.8        
## [45] irlba_2.3.3                 compiler_4.0.2             
## [47] pkgdown_1.6.1.9000          GenomeInfoDb_1.25.11       
## [49] rsvd_1.0.3                  systemfonts_0.3.1          
## [51] rlang_0.4.7                 RCurl_1.98-1.2             
## [53] BiocNeighbors_1.7.0         rjson_0.2.20               
## [55] SingleCellExperiment_1.11.7 circlize_0.4.10            
## [57] igraph_1.2.5                bitops_1.0-6               
## [59] rmarkdown_2.3               R6_2.4.1                   
## [61] rdetools_1.0                zoo_1.8-8                  
## [63] knitr_1.29                  clue_0.3-57                
## [65] rprojroot_1.3-2             ragg_0.3.1                 
## [67] shape_1.4.5                 desc_1.2.0                 
## [69] stringi_1.5.3               Rcpp_1.0.5                 
## [71] scran_1.17.17               png_0.1-7                  
## [73] xfun_0.17</code></pre>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>Accessible via ProteomXchange<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Soroor Hediyeh-zadeh.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
